lib/ui.coffee

```diff
@@ -15,14 +17,15 @@ module.exports =
     @subs.add @client.onDetached =>
       @ink?.Result.invalidateAll()
 
-    @client.handle progress: (p) =>
-      @progress?.progress = p
+    @client.handle 'progress!': (t, p) => @progress[t] p
 
   deactivate: ->
     @subs.dispose()
+    @progress.clear()
 
   consumeInk: (@ink) ->
     @views.ink = @ink
+    @progress.ink = @ink
 
-    @subs.add @client.onWorking => @progress = @ink.progress.push()
-    @subs.add @client.onDone => @progress?.destroy()
+    @subs.add @client.onWorking => @progress.add progress: -1, id: 'indet'
+    @subs.add @client.onDone => @progress.clear()
```

Comment:
This will break things if you set up a background task via `@schedule for ...` (which is actually a pretty cool use case for this feature)

I would also put the progress-specific client handler and listeners in progress.coffee itself.